site/git-gpg.md

```diff
@@ -0,0 +1,235 @@
+      <n>  = key expires in n days
+      <n>w = key expires in n weeks
+      <n>m = key expires in n months
+      <n>y = key expires in n years
+  Key is valid for? (0)
+  ```
+
+5. Verify your selection of choices and accept (`y` and `Enter`)
+
+6. Enter your user ID information, it is recommended to set the email
+   address to the same address as the daemon uses for Git operations.
+
+7. **Do not enter a passphrase**, as Flux will be unable to sign with a
+   passphrase protected private key, instead, keep it in a secure place.
+
+8. You can validate the public and private keypair were created with
+   success by running:
+
+   ```sh
+   $ gpg --list-secret-keys --keyid-format long <email address>
+   sec   rsa2048/4521D36E33A84444 2019-03-25 [SC]
+         2E0C15DD5B3578BD1AC982A14521D36E33A84444
+   uid                 [ultimate] Fluxd <flux@weave.works>
+   ssb   rsa2048/FBDA462F10B05292 2019-03-25 [E]
+   ```
+
+## Importing a GPG signing key
+
+Any file found in the configured `--git-gpg-key-import` path will be
+imported into GPG; therefore, by volume-mounting a key into that
+directory it will be made available for use by Flux.
+
+1. Retrieve the key ID (second row of the `sec` column):
+
+   ```sh
+   $  gpg --list-secret-keys --keyid-format long <email address>
+   sec   rsa2048/4521D36E33A84444 2019-03-25 [SC]
+         2E0C15DD5B3578BD1AC982A14521D36E33A84444 # key id
+   uid                 [ultimate] Fluxd <flux@weave.works>
+   ssb   rsa2048/FBDA462F10B05292 2019-03-25 [E]
+   ```
+
+2. Export the public and private keypair from your local GPG keyring
+   to a Kubernetes secret with `--export-secret-keys <key id>`:
+
+   ```sh
+   $ gpg --export-secret-keys 4521D36E33A84444 |
+     kubectl create secret generic flux-gpg-keys --from-file=flux.asc=/dev/stdin --dry-run -o yaml
+   apiVersion: v1
+   data:
+     flux.asc: <base64 string>
+   kind: Secret
+   metadata:
+     creationTimestamp: null
+     name: flux-gpg-keys
+
+3. Adapt your Flux deployment to mount the secret and enable the
+   signing of commits:
+
+   ```yaml
+   spec:
+     template:
+        spec:
+          volumes:
+          - name: gpg-keys
+              secret:
+                secretName: flux-gpg-keys
+                defaultMode: 0400
+          containers:
+          - name: flux
+            volumeMounts:
+            - name: gpg-keys
+              mountPath: /root/gpg-import
+              readOnly: true
+            args:
+            - --git-gpg-key-import=/root/gpg-import
+            - --git-signing-key=FBDA462F10B05292 # key id
+   ```
+
+   or set the `gpgKeys.secretName` in your Helm `values.yaml` to
+   `gpg-keys`, and `signingKey` to your `<key id>`.
+
+> **Note:** Flux *does not* recursively scan a given directory but does
+understand symbolic links to files.
+
+> **Note:** Flux will automatically add any imported key to the GnuPG
+  trustdb. This is required as git will otherwise not trust signatures
+  made with the imported keys.
+
```

Comment:
It'd be good to explain how to check that the signatures are working. I used `git log --show-signature` after getting fluxd to make a commit to the repo.